packages/pwa-kit-runtime/src/utils/ssr-config.client.js

```diff
@@ -12,5 +12,22 @@
  */
 /* istanbul ignore next */
 export const getConfig = () => {
+    // NOTE: window.__CONFIG__ isn't always available!
+    // The global object is populated after the pwa-kit-react-sdk hydrates data on the client side.
+    // To avoid unexpected behavior, if hydration is not complete, we parse the script manually here
+    // so getConfig() always returns the expected object, even if it is called in the global scope.
+    if (!window.__CONFIG__ && typeof document !== 'undefined') {
+        try {
+            const data = JSON.parse(document.getElementById('mobify-data').innerHTML)
+
+            // Set all globals sent from the server on the window object.
+            Object.entries(data).forEach(([key, value]) => {
+                window[key] = value
+            })
+        } catch (error) {
+            console.error('Unable to parse server-side rendered config.')
+            console.error(error)
+        }
+    }
```

Comment:
As you mentioned this is leaking internal details into the runtime specifically those from the react-sdk. This isn't ideal, but not horrible. But if, like you said, it might not be required in this implementation, then I would take it out.

Reply:
reverted